debian/patches/pop_gpu_offloading.patch

```diff
@@ -0,0 +1,219 @@
++                                         NULL,
++                                         &error);
++  if (!proxy)
++    {
++      g_debug ("Unable to connect to net.hadess.SwitcherooControl: %s", error->message);
++      return FALSE;
++    }
++
++  dualgpu_variant = g_dbus_proxy_get_cached_property (proxy, "HasDualGpu");
++
++  if (!dualgpu_variant)
++    {
++      g_debug ("Unable to retrieve net.hadess.SwitcherooControl.HasDualGpu property, the daemon is likely not running");
++      return FALSE;
++    }
++
++  ret = g_variant_get_boolean (dualgpu_variant);
++
++  if (ret)
++    g_debug ("Dual-GPU machine detected");
++
++  return ret;
++}
++
++static const char *
++get_graphics_mode (void)
++{
++  g_autoptr(GDBusProxy) proxy = NULL;
++  g_autoptr(GError) error = NULL;
++  g_autoptr(GVariant) mode = NULL;
++  const char *ret;
++
++  proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
++                                         G_DBUS_PROXY_FLAGS_NONE,
++                                         NULL,
++                                         "com.system76.PowerDaemon",
++                                         "/com/system76/PowerDaemon",
++                                         "com.system76.PowerDaemon",
++                                         NULL,
++                                         &error);
++  if (!proxy)
++    {
++      g_debug ("Unable to connect to com.system76.PowerDaemon: %s", error->message);
++      return NULL;
++    }
++
++  mode = g_dbus_proxy_call_sync (proxy,
++                                 "GetGraphics",
++                                 NULL,
++                                 0,
++                                 -1,
++                                 NULL,
++                                 &error);
++  if (!mode)
++    {
++      g_error ("Unable to determine graphics mode: %s", error->message);
++      return NULL;
++    }
++
++  g_variant_get (mode, "(s)", &ret, NULL);
++  g_debug ("Graphics mode: %s", ret);
++  return ret;
++}
++
++static void
++get_offloading_enabled (CcApplicationsPanel *self,
++                        GAppInfo            *info,
++                        gboolean            *set,
++                        gboolean            *enabled)
++{
++  const char *mode;
++  gboolean discrete_gpu = FALSE;
++
++  mode = get_graphics_mode ();
++  *set = g_strcmp0(mode, "hybrid") == 0;
++
++  if (self->gpu_offloading_settings)
++    discrete_gpu = g_settings_get_boolean (self->gpu_offloading_settings, "offload");
++
++  // Check the Desktop Entry to see if we should enable offloading.
++  if (!discrete_gpu && G_IS_DESKTOP_APP_INFO (info))
++    {
++      GDesktopAppInfo *desktop_info = G_DESKTOP_APP_INFO (info);
++
++      if (desktop_info && g_desktop_app_info_has_key (desktop_info, "X-KDE-RunOnDiscreteGpu")) {
++          discrete_gpu = g_desktop_app_info_get_boolean (desktop_info, "X-KDE-RunOnDiscreteGpu");
++      } else {
++          discrete_gpu = FALSE;
++      }
```

Comment:
Seems this could be simplified to

```c
discrete_gpu = desktop_info
    && g_desktop_app_info_has_key (desktop_info, "X-KDE-RunOnDiscreteGpu")
    && g_desktop_app_info_get_boolean (desktop_info, "X-KDE-RunOnDiscreteGpu");
```